enclave_manager/avalon_enclave_manager/epid_attestation.py

```diff
@@ -0,0 +1,198 @@
+        if not enclave_info.is_sgx_simulator():
+            logger.info("Running in Intel SGX HW mode")
+            logger.debug("posting verification to IAS")
+            ias_nonce = '{0:032X}'.format(random.getrandbits(128))
+            response = self._ias.post_verify_attestation(
+                quote=signup_data['enclave_quote'], nonce=ias_nonce)
+            logger.debug("posted verification to IAS")
+
+            # check verification report
+            if not self._ias.verify_report_fields(
+                    signup_data['enclave_quote'],
+                    response['verification_report']):
+                logger.debug(
+                    "last error: " + self._ias.last_verification_error())
+                if self._ias.last_verification_error() == "GROUP_OUT_OF_DATE":
+                    logger.warning(
+                        "failure GROUP_OUT_OF_DATE " +
+                        "(update your BIOS/microcode!!!) keep going")
+                else:
+                    logger.error("invalid report fields")
+                    return None
+            # ALL checks have passed
+            logger.info("report fields verified")
+
+            # Now put the proof data into the dictionary
+            signup_info['proof_data'] = json.dumps({
+                'verification_report': response['verification_report'],
+                'ias_report_signature': response['ias_signature'],
+                'ias_report_signing_certificate': response['ias_certificate']
+            })
+            # Grab the EPID pseudonym and put it in the enclave-persistent
+            # ID for the signup info
+            verification_report_dict = \
+                json.loads(response['verification_report'])
+            signup_info['enclave_persistent_id'] = \
+                verification_report_dict.get('epidPseudonym')
+
+        return signup_info
+
+    # -----------------------------------------------------------------
+
+    def init_enclave_info(self, signed_enclave, persisted_sealed_data,
+                          num_of_enclave):
+        """
+        Parameters :
+            @param signed_enclave - The enclave binary read from filesystem
+            @param persisted_sealed_data - persisted sealed data file path
+            @param num_of_enclave - number of enclave to create
+        Returns :
+            @returns True on success False on failure
+        """
+        if not self._tcf_enclave_info:
+            self._tcf_enclave_info = enclave_info.tcf_enclave_info(
+                signed_enclave, self._config['spid'], persisted_sealed_data,
+                int(num_of_enclave))
+
+        sig_rl_updated = False
+        while not sig_rl_updated:
+            try:
+                self._update_sig_rl()
+                sig_rl_updated = True
+            except (SSLError, Timeout, HTTPError) as e:
+                logger.warning(
+                    "Failed to retrieve initial sig rl from IAS: %s", str(e))
```

Comment:
Did you mean "signal" instead "sig rl"? 
I don't understand what a "sig rl" is.

Reply:
We can mention "signature revocation list"